### PR TITLE
implement option for uploading local test data

### PR DESCRIPTION
**Pull request by martenson**
this option needs the target Galaxy to enable config option `allow_path_paste`

### CODE REVIEW

File: galaxy/tools/verify/interactor.py [hunk 1]
```diff
@@ -297,16 +297,20 @@ def stage_data_async(self, test_data, history_id, tool_id, async=True):
                 "files_0|NAME": name,
                 "files_0|type": "upload_dataset",
             })
-            # TODO: Option to upload by path since we are getting the paths from Galaxy now it makes more
-            # sense to move this there.
-            files = {
-                "files_0|file_data": open(file_name, 'rb')
-            }
+            files = {}
```
Comment (jmchilton) [inline galaxy/tools/verify/interactor.py]: same change at line 285 above

File: galaxy/tools/verify/interactor.py [hunk 2]
```diff
@@ -559,6 +563,9 @@ def _post(self, path, data=None, files=None, key=None, admin=False, anon=False):
         params, data = self.__inject_api_key(data=data, key=key, admin=admin, anon=anon)
         # no params for POST
         data.update(params)
+        log.debug("POST DEBUG")
+        log.debug(data)
+        log.debug(files)
```
Comment (jmchilton) [inline galaxy/tools/verify/interactor.py]: drop?

File: galaxy/tools/verify/script.py [hunk 3]
```diff
@@ -93,6 +93,7 @@ def _arg_parser():
     parser.add_argument('-u', '--galaxy-url', default="http://localhost:8080", help='Galaxy URL')
     parser.add_argument('-k', '--key', default=None, help='Galaxy User API Key')
     parser.add_argument('-a', '--admin-key', default=None, help='Galaxy Admin API Key')
+    parser.add_argument('--allow_path_paste', default=False, action="store_true", help='This requires Galaxy-side config option with the same name enabled. Allows for fetching test data locally. Only for admins.')
```
Comment (jmchilton) [inline galaxy/tools/verify/script.py]: ``force_path_paste`` would be better IMO.

**Comment by martenson** [comment]
the rest of the failing lints were not cause by me

**Comment by mvdbeek** [comment]
What is this being used for ? just tool tests ? In which case should we go the easy route and revert it ?

**Comment by martenson** [comment]
@mvdbeek I suspect so? I think it was aimed to resolve bullet number 3 at https://github.com/galaxyproject/galaxy/issues/5651

**Comment by jmchilton** [comment]
I'm not sure I agree it should be reverted. I feel like pasting paths is more efficient and robust. It is closer to the final solution I'd like to see of a special tool-test uri recognized by upload tools. I know this is a lot of complexity but I'm not sure I like not seeing progress toward something better then downloading and uploading the file on the client when the file is already on the server.

**Comment by jmchilton** [comment]
None of that is to say https://github.com/galaxyproject/galaxy/pull/7160/files isn't needed and great - because it is and it should be the new default for now.

**Pull request merged**